bitnami/spark/values-production.yaml

```diff
@@ -0,0 +1,237 @@
+#   imagePullSecrets:
+#     - myRegistryKeySecretName
+
+## Bitnami Spark image version
+## ref: https://hub.docker.com/r/bitnami/spark/tags/
+##
+image:
+  registry: docker.io
+  repository: 'bitnami/spark'
+  tag: 2.4.3
+  ## Specify a imagePullPolicy
+  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
+  ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
+  ##
+  pullPolicy: IfNotPresent
+
+  ## Pull secret for this image
+  # pullSecrets:
+  #   - myRegistryKeySecretName
+
+
+## Spark Components configuration
+##
+master:
+  ## Spark master specific configuration
+  webPort: 8080
+  clusterPort: 7077
+
+  ## Set the master daemon memory limit.
+  # daemonMemoryLimit:
+
+  ## Use a string to set the config options for in the form "-Dx=y"
+  # configOptions:
+
+  ## Set to true if you would like to see extra information on logs
+  ## It turns BASH and NAMI debugging in minideb
+  ## ref:  https://github.com/bitnami/minideb-extras/#turn-on-bash-debugging
+  debug: false
+  ## StatefulSet controller supports automated updates. There are two valid update strategies: RollingUpdate and OnDelete
+  ## ref: https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#updating-statefulsets
+  ##
+  updateStrategy: RollingUpdate
+  ## Number of replicas
+  replicaCount: 1
+  ## Kubernetes Security Context
+  ## https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  ##
+  securityContext:
+    enabled: true
+    fsGroup: 1001
+    runAsUser: 1001
+  ## Node labels for pod assignment
+  ## Ref: https://kubernetes.io/docs/user-guide/node-selection/
+  ##
+  nodeSelector: {}
+  ## Tolerations for pod assignment
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  ##
+  tolerations: []
+  ## Affinity for pod assignment
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+  ##
+  affinity: {}
+  ## Configure resource requests and limits
+  ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+  ##
+  resources:
+  #  limits:
+  #    cpu: 200m
+  #    memory: 1Gi
+  #  requests:
+  #    memory: 256Mi
+  #    cpu: 250m
+
+  ## Configure extra options for liveness and readiness probes
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes)
+  livenessProbe:
+    enabled: true
+    initialDelaySeconds: 180
+    periodSeconds: 20
+    timeoutSeconds: 5
+    failureThreshold: 6
+    successThreshold: 1
+
+  readinessProbe:
+    enabled: true
+    initialDelaySeconds: 30
+    periodSeconds: 10
+    timeoutSeconds: 5
+    failureThreshold: 6
+    successThreshold: 1
+
+worker:
+  ## Spark worker specific configuration
+
+  webPort: 8081
+  ## Set to true to use a custom cluster port instead of a random port.
+  customClusterPort: false
+  # clusterPort:
+
+  ## Set the daemonMemoryLimit as the daemon max memory
```

Comment:
Does the master require this set of values?

Reply:
Not sure about what are you refering to. Ports are configurable. In the master, the cluster port has a default value, while in the worker it is random by default, but you can provide a static port.